tmt/steps/execute/internal.py

```diff
@@ -193,39 +192,39 @@ def __init__(self, **kwargs: Any):
     def _test_environment(
             self,
             *,
-            test: Test,
-            guest: Guest,
+            invocation: TestInvocation,
             extra_environment: Optional[EnvironmentType] = None,
             logger: tmt.log.Logger) -> EnvironmentType:
         """ Return test environment """
 
         extra_environment = extra_environment or {}
 
-        data_directory = self.data_path(test, guest, full=True, create=True)
+        data_directory = invocation.data_path(full=True, create=True)
 
         environment = extra_environment.copy()
-        environment.update(test.environment)
+        environment.update(invocation.test.environment)
         assert self.parent is not None
         assert isinstance(self.parent, tmt.steps.execute.Execute)
 
         environment['TMT_TEST_PIDFILE'] = str(
             effective_pidfile_root() / TEST_PIDFILE_FILENAME)
         environment['TMT_TEST_PIDFILE_LOCK'] = str(
             effective_pidfile_root() / TEST_PIDFILE_LOCK_FILENAME)
-        environment["TMT_TEST_NAME"] = test.name
+        environment["TMT_TEST_NAME"] = invocation.test.name
         environment["TMT_TEST_DATA"] = str(data_directory / tmt.steps.execute.TEST_DATA)
-        environment['TMT_TEST_SERIAL_NUMBER'] = str(test.serial_number)
+        environment['TMT_TEST_SERIAL_NUMBER'] = str(invocation.test.serial_number)
         environment["TMT_TEST_METADATA"] = str(
             data_directory / tmt.steps.execute.TEST_METADATA_FILENAME)
         environment["TMT_REBOOT_REQUEST"] = str(
             data_directory / tmt.steps.execute.TEST_DATA / TMT_REBOOT_SCRIPT.created_file)
         # Set all supported reboot variables
         for reboot_variable in TMT_REBOOT_SCRIPT.related_variables:
-            environment[reboot_variable] = str(test._reboot_count)
+            environment[reboot_variable] = str(invocation.test._reboot_count)
 
         # Add variables the framework wants to expose
-        environment.update(test.test_framework.get_environment_variables(
-            self, test, guest, logger))
+        environment.update(
+            invocation.test.test_framework.get_environment_variables(
+                invocation, logger))
```

Comment:
Seeing this (invocation object passed in, even if called from an invocation instance), maybe invocation should be later available to the objects, so we do not need to pass it again?

Reply:
Looking at `get_environment_variables()`, passing just `invocation.data_path(full=True)` instead of the whole `invocation` might be enough.

Reply:
I would stick with invocation rather than just data path - in the future, a `pytest` framework may be interested in getting more, and I would like to avoid picking bits from invocation, that would be the opposite of what I'm trying to do :) Pass it in, let the code decide what it'd use from it, don't force it to live with just the data path. It's for the same price :)

> Seeing this (invocation object passed in, even if called from an invocation instance), maybe invocation should be later available to the objects, so we do not need to pass it again?

I'm not sure I follow. This line is not called from an invocation instance, as the invocation class performs no actions on its own. The code needs to pass the invocation to the framework as there is a single test instance we share among different threads when running the same test multiple times at one moment. Therefore the code needs to pass the invocation to the framework rather than the framework finding it attached to the test - a single test instance, shared.

Reply:
@happz's comments sound reasonable to me.

Reply:
thx for the explanation